pkg/controller/bookkeepercluster/bookie.go

```diff
@@ -205,33 +205,62 @@ func makeBookiePodSpec(bk *v1alpha1.BookkeeperCluster) *corev1.PodSpec {
 
 func createVolumeMount(ledgerDirs []string, journalDirs []string, indexDirs []string) []corev1.VolumeMount {
 	var volumeMounts []corev1.VolumeMount
-	for i, ledger := range ledgerDirs {
-		name := LedgerDiskName + strconv.Itoa(i)
+	if len(ledgerDirs) > 0 && !strings.EqualFold(ledgerDirs[0], "/bk/ledgers") {
```

Comment:
This if block should be executed only if `ledgerDirs` contains a single element. No string comparison would be needed here

Reply:
Done.